corehq/apps/users/management/commands/migrate_reprs.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+from corehq.apps.users.models import UserHistory
+from corehq.apps.users.util import cached_user_id_to_username
+
+from corehq.apps.users.util import SYSTEM_USER_ID
+from corehq.util.log import with_progress_bar
+
+
+class Command(BaseCommand):
+    help = "Adds user_repr and changed_by_repr to UserHistory log if not already present"
+
+    def handle(self, **options):
+        records = UserHistory.objects.filter(Q(user_repr__isnull=True) | Q(changed_by_repr__isnull=True))
+        for user_history in with_progress_bar(records.order_by('pk').iterator(), records.count()):
```

Comment:
This might run into scale issues.  Unfortunately, `.iterator()` doesn't actually do anything particularly clever - it still evaluates the entire queryset in one go in our case.  See [the docs](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#iterator) - we use PSQL, but with server side cursors disabled because of transaction pooling.

An alternative is to fetch just IDs from the query, then pull rows in chunks, like here:
https://github.com/dimagi/commcare-hq/blob/17511d9c3d23905698679830c4147d8f34b6afe7/corehq/apps/users/management/commands/add_download_reports_permission.py#L13-L18

Reply:
Ha, looks like we have a wrapper for that:
https://github.com/dimagi/commcare-hq/blob/17511d9c3d23905698679830c4147d8f34b6afe7/corehq/util/queries.py#L102-L102

Reply:
That's convenient.